smartsim/_core/_cli/utils.py

```diff
@@ -65,26 +66,51 @@ def pip_install(
     Currently only Torch shared libraries are re-used for the build
     """
     # form pip install command
-    cmd = ["python", "-m", "pip", "install"]
-    cmd.extend(packages)
+    cmd = [sys.executable, "-m", "pip", "install"]
+
     if end_point:
-        cmd.extend(["-f", end_point])
+        cmd.extend(("-f", str(end_point)))
+
+    packages.sort()
+    pkg_str_list = "\n\t".join(packages)
+
+    if verbose:
+        logger.info(f"Attempting to Install Packages:\n\t{pkg_str_list}")
+    for package in packages:
+        proc = subprocess.Popen(
```

Comment:
the content of the install/uninstall loops are pretty similar. you could yank the bodies out & parameterize them (e.g. `cmd: str, pkg_name: str, action: str = "Install",`)... that would give you the ability to mock the actual installation and test that the build does try to install the packages you expect